Assets/Scripts/EnemyController.cs

```diff
@@ -49,24 +69,21 @@ void FixedUpdate()
     public void ChangeHealth(int healthLost)
     {
         currentHealth += healthLost;
-        if(currentHealth <= 0)
-        {
-            Die();
-        }
+
+        if (currentHealth <= 0) { Die(); }
     }
 
     void Die()
     {
+        if (player != null)
+        {
+            player.enemyHash.Remove(thisTransform);
+            inHash = false;
+        }
         Object.Destroy(this.gameObject);
     }
 
-    bool IsWithinCamera()
-    {
-        if (GetComponent<Renderer>().isVisible)
-        {
-            return true;
-        }else return false;
-    }
+    bool IsWithinCamera() => (GetComponent<Renderer>().isVisible) ? true : false;
```

Comment:
`GetComponent<Renderer>()` should be set to a variable in the `Awake` method or the `Start` method. That's not really related to your changes, though.

Reply:
That's alright, if it's for the better, I'm always down to change whatever.

Reply:
Also "transform can be used instead of setting and using thisTransform ([documentation](https://docs.unity3d.com/ScriptReference/Component-transform.html))." this is true of all components.